ports/servoshell/egl/log.rs

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+//! Helper Module to redirect stdout/stderr to the logging sink
+
+use std::os::fd::{AsRawFd, IntoRawFd, RawFd};
+use std::thread;
+
+use log::{debug, error, info, warn};
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub(crate) enum LogRedirectError {
+    CreatePipeFailed(nix::Error),
+    RedirectToPipeFailed(nix::Error),
+}
+
+/// Redirect stdout and stderr to the logging system
+pub(crate) fn redirect_stdout_and_stderr() -> Result<(), LogRedirectError> {
+    fn log_raw_msg(raw_msg: &[u8]) {
+        if let Ok(utf8_msg) = std::str::from_utf8(raw_msg) {
```

Comment:
since this can fail, maybe better to return error from log_raw_msg?

Reply:
I put some comments on how we could handle likely errors in the `else` case, but I was thinking maybe it makes sense to first see in practice if we even hit such cases.
I still have to test this PR in practice though - since I updated my phone to the latest beta version, servo keeps crashing in some spidermonkey code and I'm still investigating that.

Reply:
I got back to this PR today, and tested it. The redirection seems to work as expected on OH. I also couldn't see any warning messages from the error case here.